packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
    * @default false
    */
   isDisabled?: boolean;
+  setActionable?: React.Dispatch<React.SetStateAction<boolean>>;
   onClick?: React.MouseEventHandler<HTMLElement>;
 };
 
-/**
- * Tile component that displays content and actions in a card-like format.
- * @example
- * <Tile
- *  name='Tile name'
- *  description='Tile description that takes upto 3 lines'
- *  metadata={<TagContainer><Tag variant='basic'>Tag 1</Tag></TagContainer>}
- *  thumbnail='/url/to/image.jpg'
- *  badge={<Badge backgroundColor='blue'>Badge label</Badge>}
- *  buttons={[<Button>Button 1</Button>, <Button>Button 2</Button>]}
- *  moreOptions={[<MenuItem>Item 1</MenuItem>, <MenuItem>Item 2</MenuItem>]}
- *  leftIcon={<IconButton><SvgInfo /></IconButton>}
- *  rightIcon={<IconButton><SvgStar /></IconButton>}
- *  isSelected={true}
- *  isNew={false}
- * />
- */
-export const Tile = Object.assign(
-  React.forwardRef((props, forwardedRef) => {
-    const {
-      className,
-      name,
-      description,
-      metadata,
-      thumbnail,
-      buttons,
-      leftIcon,
-      rightIcon,
-      badge,
-      isNew,
-      isSelected,
-      moreOptions,
-      variant = 'default',
-      children,
-      isActionable: isActionableProp,
-      status,
-      isLoading = false,
-      isDisabled = false,
-      onClick,
-      ...rest
-    } = props;
-
-    const [isMenuVisible, setIsMenuVisible] = React.useState(false);
-    const showMenu = React.useCallback(() => setIsMenuVisible(true), []);
-    const hideMenu = React.useCallback(() => setIsMenuVisible(false), []);
-    const [localActionable, setLocalActionable] =
-      React.useState(isActionableProp);
-
-    const isActionable = isActionableProp ?? localActionable;
-
-    const tileName = (
-      <Box className='iui-tile-name'>
-        <TitleIcon
-          isLoading={isLoading}
-          isSelected={isSelected}
-          isNew={isNew}
-          status={status}
-        />
-
-        <Box as='span' className='iui-tile-name-label'>
-          {isActionable && onClick ? (
-            <LinkAction
-              as='button'
-              onClick={!isDisabled ? onClick : undefined}
-              aria-disabled={isDisabled}
-            >
-              {name}
-            </LinkAction>
-          ) : (
-            name
-          )}
-        </Box>
-      </Box>
-    );
+const TileComponent = React.forwardRef((props, ref) => {
+  const {
+    className,
+    status,
+    variant,
+    isNew,
+    isSelected,
+    isLoading,
+    isActionable: isActionableProp,
+    isDisabled,
+    onClick,
```

Comment:
cool, so lets remove `onClick` from context, and remove any special handling for `onClick`, then update any leftover actionable examples to explicitly use `Tile.Action` instead of `isActionable` prop.

should we also remove `isActionable` prop?

Reply:
@mayank99 I believe this is what you mean? I moved the action logic out of `Tile.NameLabel` into `Tile.Action`, removing the `onClick` context, so now custom `onClick` if any is passed into `Tile.Action` instead (also pushed):

![image](https://github.com/iTwin/iTwinUI/assets/22462084/18d46ffc-906d-4416-b513-e324f25b8aef)

Reply:
yes, basically that. You're missing rest props though.

Also, we can set default `as` prop based on `href` vs `onClick`, similar to how it's done in `Alert.Action`

https://github.com/iTwin/iTwinUI/blob/ab07a844f5c2bca1c255adc5502e572c34fc8ede/packages/itwinui-react/src/core/Alert/Alert.tsx#L94-L102

Reply:
> should we also remove `isActionable` prop?

I don't remember if there is a use-case for actionable tile without click action. Maybe users could want that hover effect when clicking on more options menu? We could mark that as deprecated and keep only for classname toggle.

Reply:
marked @deprecated